src/midi/soundcanvas.cpp

```diff
@@ -280,10 +280,90 @@ static void setup_filter(MixerChannelPtr& channel, const bool filter_enabled)
 	}
 }
 
+#if defined(WIN32)
+
+static std::deque<std_fs::path> get_platform_rom_dirs()
+{
+	return {
+	        get_config_dir() / DefaultSoundcanvasRomsDir,
+	        "C:\\soundcanvas-rom-data\\",
+	};
+}
+
+#elif defined(MACOSX)
+
+static std::deque<std_fs::path> get_platform_rom_dirs()
+{
+	return {
+	        get_config_dir() / DefaultSoundcanvasRomsDir,
+	        resolve_home("~/Library/Audio/Sounds/SoundCanvas-Roms/"),
+	        "/usr/local/share/soundcanvas-rom-data/",
+	        "/usr/share/soundcanvas-rom-data/",
+	};
+}
+#else
+
+static std::deque<std_fs::path> get_platform_rom_dirs()
+{
+	// First priority is user-specific data location
+	const auto xdg_data_home = get_xdg_data_home();
+
+	std::deque<std_fs::path> dirs = {
+	        xdg_data_home / "dosbox" / DefaultSoundcanvasRomsDir,
+	        xdg_data_home / "soundcanvas-rom-data",
+	};
+
+	// Second priority are the $XDG_DATA_DIRS
+	for (const auto& data_dir : get_xdg_data_dirs()) {
+		dirs.emplace_back(data_dir / "soundcanvas-rom-data");
+	}
+
+	// Third priority is $XDG_CONF_HOME, for convenience
+	dirs.emplace_back(get_config_dir() / DefaultSoundcanvasRomsDir);
+
+	return dirs;
+}
+
+#endif
+
+static std::deque<std_fs::path> get_rom_dirs()
+{
+	// Get potential ROM directories from the environment and/or system
+	auto rom_dirs = get_platform_rom_dirs();
+
+	// Get the user's configured ROM directory; otherwise use 'mt32-roms'
+	std_fs::path selected_romdir = get_soundcanvas_section()->GetString("soundcanvas_rom_dir");
+
+	if (selected_romdir.empty()) { // already trimmed
+		selected_romdir = DefaultSoundcanvasRomsDir;
+	}
+
+	// Make sure we search the user's configured directory first
+	rom_dirs.emplace_front(resolve_home(selected_romdir.string()));
+	return rom_dirs;
+}
+
 MidiDeviceSoundCanvas::MidiDeviceSoundCanvas()
 {
 	using namespace SoundCanvas;
 
+	// Before attempting to load any plugins, we must set an environment
+	// variable for the ROM directory if possible. Note, on POSIX systems 
```

Comment:
Maybe soften the wording a bit: 

> Before attempting to load any plugins, we set the ROM directory location hint via the SOUNDCANVAS_ROM_DIR.

Minor thing, just I kept the coupling to the absolute minimum between Staging and the plugin 😄